File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-1.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

public class Solution {    
    public ListNode MergeKLists(ListNode[] lists) {
        PriorityQueue<ListNode, int> pq = new();

        if (lists.Length == 0)
            return null;

        foreach (ListNode node in lists)
        {
            ListNode copy = node;
            while (copy != null)
            {
                pq.Enqueue(copy, copy.val);
                copy = copy.next;
            }
        } 

        if (pq.Count == 0)
            return null;

        ListNode head = pq.Dequeue();
        ListNode curr = head;

        while (pq.Count > 0)
        {
            curr.next = pq.Dequeue();
            curr = curr.next;
        }

        curr.next = null;

        return head;
    }
}

File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        
        dummy = ListNode()
        node = dummy
            
        while list1 != None or list2 != None:
            if list1 is None:
                node.next = list2
                return dummy.next


            if list2 is None:
                node.next = list1
                return dummy.next

            if list1.val <= list2.val:
                node.next = list1
                list1 = list1.next
            else:
                node.next = list2
                list2 = list2.next
            node = node.next

        return dummy.next